src/test/groovy/net/rubygrapefruit/platform/file/FileEventFunctionsOverflowTest.groovy

```diff
@@ -79,7 +81,45 @@ class FileEventFunctionsOverflowTest extends AbstractFileEventFunctionsTest {
         executorService.shutdown()
     }
 
-    private boolean expectOverflow(BlockingQueue<RecordedEvent> eventQueue = this.eventQueue, int timeoutValue, TimeUnit timeoutUnit) {
+    def "handles Java-side event queue overflowing"() {
+        given:
+        def singleElementQueue = new ArrayBlockingQueue<FileWatchEvent>(1)
+        def firstFile = new File(rootDir, "first.txt")
+        def secondFile = new File(rootDir, "second.txt")
+
+        startWatcher(singleElementQueue, rootDir)
+
+        when:
+        createNewFile(firstFile)
+        waitForChangeEventLatency()
+
+        then:
+        singleElementQueue.peek().type == CREATED
+        singleElementQueue.peek().path == firstFile.absolutePath
+
+        when:
+        createNewFile(secondFile)
+        waitForChangeEventLatency()
+
+        then:
+        // We still have the notification in there about the first file
+        singleElementQueue.peek().type == CREATED
+        singleElementQueue.peek().path == firstFile.absolutePath
+
+        when:
+        // Wait for the next event to time out and be replaced with an overflow event
+        Thread.sleep(1000)
```

Comment:
Can we extract this constant or the sleep into a method?

Reply:
Won't be needed if we don't wait on `offer()`.